image_writers_guide/guidelines.adoc

```diff
@@ -5,143 +5,186 @@
 
-Keep in mind that even with persisted volumes, each instance of your image will have its own volume, the filesystem is not shared between instances.  This means the volume cannot be used to share state in a cluster.
+////
+For more information see https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/services.md[this documentation].  (NOTE to docs team:  this link should really go to something in the openshift docs once we have it)
+////
 
+*Providing Common Libraries*
 
+For images that are intended to run application code provided by a third party, ensure that your image contains commonly used libraries for your platform. In particular, provide database drivers for common databases used with your platform. For example, provide JDBC drivers for MySQL and PostgreSQL if you are creating a Java framework image. Doing so prevents the need for common dependencies to be downloaded during application assembly time, speeding up application image builds. It will also simplify the work required by application developers to ensure all of their dependencies are met.
 
-== OpenShift Specific Recommendations
-=== Enable images for Source-To-Image
-For images that are intended to run application code provided by a third party (e.g. a Ruby image designed to run Ruby coded provided by a developer), you should enable your image to work with the https://github.com/openshift/source-to-image[Source to Image] (STI) build tool.  It is a framework which makes it easy to write images that take application source code as an input, and produce a new image that will run the assembled application as output.  For example, this https://github.com/openshift/wildfly-8-centos[Wildfly] image defines STI scripts which will run a `maven` build on a java source repository and copy the resulting war file into the Wildfly deployments directory.  The resulting image will now automatically start Wildfly with the application running.
+[[using-env-vars]]
+*Using Environment Variables for Configuration*
 
-For more details about how to write STI scripts for your image, see the https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md[STI documentation].
+Users of your image should be able to configure it without having to create a downstream image based on your image. This means that the runtime configuration should be handled using environment variables.
+For a simple configuration, the running process can consume the environment variables directly. For a more complicated configuration or for runtimes which do not support this, configure the runtime by defining a template configuration file which is processed during startup. During this processing, values supplied using environment variables can be substituted into the configuration file or used to make decisions about what options to set in the configuration file.
```

Comment:
[Suggestion]
For a more complicated configuration or for runtimes which do not support this, configure the runtime by defining a template configuration file which is processed during startup.

s/which/that